django_ipam/base/models.py

```diff
@@ -34,14 +34,27 @@ def clean(self):
             if self.id != subnet["id"] and ip_network(self.subnet).overlaps(subnet["subnet"]):
                 raise ValidationError({'subnet': _('Subnet overlaps with %s') % (subnet["subnet"])})
 
-    def request_ip(self):
+    def get_first_available_ip(self):
         subnet = get_object_or_404(swapper.load_model("django_ipam", "Subnet"), pk=self.id)
         ipaddress_set = [ipaddr.ip_address for ipaddr in subnet.ipaddress_set.all()]
         for host in subnet.subnet.hosts():
             if str(host) not in ipaddress_set:
                 return str(host)
         return None
 
+    def request_ip(self, options={}):
+        subnet = get_object_or_404(swapper.load_model("django_ipam", "Subnet"), pk=self.id)
```

Comment:
isn't this a model method of `Subnet`? I don't get it:

when calling the model method, an instance of subnet is already initialized, why are you retrieving it again from the database?

Moreover, using `get_object_or_404` here is very wrong, because `get_object_or_404` is designed to be used in HTTP views, while here we are dealing with the database and if a `404` exception is raised django will simply fail and the system will return 500 internal server error.

Remove this code altogether.